### PR TITLE
enhancement: show an error message if openLink fails

### PR DESCRIPTION
**Pull request by tarunsamanta2k20**
Use catch block to show a message that opening the link does not work

https://github.com/cashubtc/eNuts/blob/main/src/util/index.ts#L178

closes: #58

### CODE REVIEW

**Comment by tarunsamanta2k20** [comment]
@KKA11010 where should i put toast?

**Review by KKA11010 — CHANGES_REQUESTED** [review]
Hello @tarunsamanta2k20 

Thanks for the PR! 💪

Yes the Toaster component should be in each screen where this openUrl is called. But I think it will not work if you write the {prompt} hook in the openUrl function.

I just use the prompt hook in the same component where openUrl is needed.

```typescript
// component where openUrl is called

Component xyz () {

const {prompt, openPromptAutoClose} ...

testFunction(){
  Try {
    OpenUrl...
  } catch (e) {
    OpenPrompt({...})
  }
}

Return <>
  ... component content ...
  <Button onPress={testFunction}>
   ...Btn Text...
  </>

  <Toaster />
</>
```

Maybe like this? But you can just play around and see if your way works

👍

**Comment by tarunsamanta2k20** [comment]
@KKA11010 i think it should be like that.

export function OpenUrl(url: string)

**Comment by KKA11010** [comment]
> @KKA11010 i think it should be like that.
> 
> export function OpenUrl(url: string)

Yes, it is already.

[https://github.com/cashubtc/eNuts/blob/main/src/util/index.ts#L174](https://github.com/cashubtc/eNuts/blob/main/src/util/index.ts#L174)

Just play around and see if it works the way you are doing it.

**Comment by tarunsamanta2k20** [comment]
> > @KKA11010 i think it should be like that.
> > export function OpenUrl(url: string)
> 
> Yes, it is already.
> 
> https://github.com/cashubtc/eNuts/blob/main/src/util/index.ts#L174
> 
> Just play around and see if it works the way you are doing it.

See 'O' is in lower case. That why it was giving error.

**Comment by KKA11010** [comment]
@tarunsamanta2k20 the code example that I provided was just pseudo code as an example.

File: src/components/Balance.tsx [hunk 1]
```diff
@@ -51,12 +53,13 @@ export default function Balance({ balance }: IBalanceProps) {
 				</Text>
 				<TouchableOpacity
 					style={styles.submitIssue}
-					onPress={() => void openUrl(repoIssueUrl)}
+					onPress={() => void openUrl(repoIssueUrl)?.catch((err: string) => openPromptAutoClose({ msg: err }) )}
```
Comment (KKA11010) [inline src/components/Balance.tsx:56]: I think err is not a string.

catch(e => openPromptAutoClose({msg: e instanceof Error ? e.message : 'Link could not be opened'})

File: src/components/ErrorScreen/ErrorDetails.tsx [hunk 2]
```diff
@@ -25,13 +29,14 @@ export function ErrorDetails(props: ErrorDetailsProps) {
 				<Txt txt={props.errorInfo?.componentStack ?? 'Error stack not available'} />
 			</ScrollView>
 			<TouchableOpacity
-				onPress={() => void openUrl(repoIssueUrl)}
+				onPress={() => void openUrl(repoIssueUrl)?.catch((err: string) => openPromptAutoClose({ msg: err }) )}
```
Comment (KKA11010) [inline src/components/ErrorScreen/ErrorDetails.tsx:32]: Same

File: src/components/screens/Lightning/modal.tsx [hunk 3]
```diff
@@ -148,7 +151,7 @@ export function InvoiceModal({ visible, invoice, mintUrl, close }: IInvoiceModal
 							txt='Pay with your LN wallet'
 							onPress={() => {
 								void (async () => {
-									await openUrl(`lightning:${invoice.decoded?.paymentRequest ?? ''}`)
+									await openUrl(`lightning:${invoice.decoded?.paymentRequest ?? ''}`)?.catch((err: string) => openPromptAutoClose({ msg: err }) )
```
Comment (KKA11010) [inline src/components/screens/Lightning/modal.tsx:154]: Same

File: src/components/screens/Lightning/payInvoice.tsx [hunk 4]
```diff
@@ -380,7 +380,7 @@ export default function PayInvoicePage({ navigation, route }: TPayLNInvoicePageP
 				{!input.length && !isKeyboardOpen &&
 					<TouchableOpacity style={{ marginVertical: 10 }} onPress={() => {
 						void (async () => {
-							await openUrl('lightning://')
+							await openUrl('lightning://')?.catch((err: string) => openPromptAutoClose({ msg: err }) )
```
Comment (KKA11010) [inline src/components/screens/Lightning/payInvoice.tsx:383]: Same

**Review by KKA11010 — CHANGES_REQUESTED** [review]
Looks great, thank you! Just a few minor changes needed i guess 😍

**Comment by KKA11010** [comment]
@tarunsamanta2k20 also please have a look at pnpm if you do not use it yet 👍

Yesterday I merged a PR from @BilligsterUser and now we are using pnpm instead of npm

File: src/components/screens/Lightning/payInvoice.tsx (hunk 4 above)
Comment (tarunsamanta2k20) [reply src/components/screens/Lightning/payInvoice.tsx:383]: Have you checked msg data type in your interface? It is string. Right?

**Review by KKA11010 — APPROVED** [review]
LGTM

**Pull request merged**